test/test_client_context.py

```diff
@@ -21,6 +21,7 @@
 
 
 class TestClientContext(unittest.TestCase):
+    @client_context.require_connection
```

Comment:
These tests are testing client_context itself so you can't use require_connection here. If this test is failing then somethign else is broken.